### PR TITLE
Add flag Find all Markdown files, Ignore ts/te tags, Automatically at the end/head of the file.

### PR DESCRIPTION
**Pull request by yiranzai**
--all                 Find all Markdown files for non-hidden folders 
--auto               Ignore ts/te tags, Automatically at the end/head of the file
 --head            The TOC is generated in the header of the file, requires --auto

### CODE REVIEW

**Review by ekalinin — COMMENTED** [review]
Thanks for the patch!

Could you, please, check my comments.
And one more thing: could you add some tests for a new functionality?

File: gh-md-toc [hunk 1]
```diff
@@ -139,43 +141,67 @@ gh_toc(){
...
+                fi
             fi
-            local ts="<\!--ts-->"
-            local te="<\!--te-->"
-            local dt=`date +'%F_%H%M%S'`
+            local dt=$(date +'%F_%H%M%S')
             local ext=".orig.${dt}"
             local toc_path="${gh_src}.toc.${dt}"
-            local toc_footer="<!-- Added by: `whoami`, at: `date` -->"
+            local toc_footer="<!-- Added by: $(whoami), at: $(date) -->"
```
[the remark this answers is not in the document]
Comment (yiranzai) [reply gh-md-toc:193]: https://github.com/koalaman/shellcheck/wiki/SC2006

File: gh-md-toc [hunk 2]
```diff
@@ -296,13 +333,13 @@ gh_toc_app() {
     if [ "$1" = '--version' ]; then
         echo "$gh_toc_version"
         echo
-        echo "os:     `lsb_release -d | cut -f 2`"
-        echo "kernel: `cat /proc/version`"
-        echo "shell:  `$SHELL --version`"
+        echo "os:     $(lsb_release -d | cut -f 2)"
```
[the remark this answers is not in the document]
Comment (yiranzai) [reply gh-md-toc:336]: https://github.com/koalaman/shellcheck/wiki/SC2006

File: gh-md-toc [hunk 3]
```diff
@@ -139,43 +141,67 @@ gh_toc(){
...
+
+                else
+                    echo "You don't have <!--ts--> or <!--te--> in your file...exiting"
+                    exit 1
+                fi
             fi
-            local ts="<\!--ts-->"
-            local te="<\!--te-->"
-            local dt=`date +'%F_%H%M%S'`
+            local dt=$(date +'%F_%H%M%S')
```
[the remark this answers is not in the document]
Comment (yiranzai) [reply gh-md-toc:190]: https://github.com/koalaman/shellcheck/wiki/SC2006

File: gh-md-toc [hunk 4]
```diff
@@ -274,19 +300,30 @@ gh_toc_get_filename() {
     echo "${1##*/}"
 }
 
+#
+# Find all Markdown files for non-hidden folders
+#
+all_md_filename=""
+gh_all_md_filename() {
```
[the remark this answers is not in the document]
Comment (yiranzai) [reply gh-md-toc:307]: You are right

File: gh-md-toc [hunk 5]
```diff
@@ -139,43 +141,67 @@ gh_toc(){
...
-             echo "or place GitHub auth token here: ${TOKEN_FILE}"
-             exit 1
+            echo "Parsing local markdown file requires access to github API"
+            echo "Error: You exceeded the hourly limit. See: https://developer.github.com/v3/#rate-limiting"
+            TOKEN_FILE="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/token.txt"
+            echo "or place GitHub auth token here: ${TOKEN_FILE}"
+            exit 1
         fi
-        local toc=`echo "$rawhtml" | gh_toc_grab "$gh_src_copy"`
+        local toc=$(echo "$rawhtml" | gh_toc_grab "$gh_src_copy")
```
[the remark this answers is not in the document]
Comment (yiranzai) [reply gh-md-toc:155]: https://github.com/koalaman/shellcheck/wiki/SC2006

**Comment by yiranzai** [comment]
I could revert to these changes, but this project seems to need a shell style checker

**Pull request closed without merging**